### PR TITLE
web: distinguish browser error from network error

### PR DESCRIPTION
**Pull request by pivotal-jamie-klassen**
fixes concourse/concourse#4409

### CODE REVIEW

**Comment by pivotal-jamie-klassen** [comment]
I think we can switch the JSON decoding with just using strings, and remove the BadUrl variant

File: web/elm/src/Message/Subscription.elm [hunk 1]
```diff
@@ -32,6 +33,17 @@ port reportIsVisible : (( String, Bool ) -> msg) -> Sub msg
...
 
+port rawHttpResponse : (Json.Encode.Value -> msg) -> Sub msg
+
+
+type RawHttpResponse
+    = Success -- ()
+    | BadUrl String -- ({"badUrl": "the-bad-url./com"})
+    | Timeout -- ({"timeout": true})
+    | NetworkError -- ({"networkError": true})
+    | BrowserError -- ({"browserError": true})
```
Comment (pivotal-jamie-klassen) [inline web/elm/src/Message/Subscription.elm]: tests would be better than comments here

File: web/elm/tests/FlySuccessTests.elm [hunk 2]
```diff
@@ -241,6 +271,19 @@ property query description assertion setup =
...
+-- subscription
+
+
+listensForTokenResponse : Property
+listensForTokenResponse setup =
+    test (setupDesc setup ++ ", listens for token response") <|
+        steps setup
+            >> Application.subscriptions
+            >> List.member Subscription.OnTokenSentToFly
+            >> Expect.equal True
```
Comment (pivotal-jamie-klassen) [inline web/elm/tests/FlySuccessTests.elm]: maybe a better 'list membership' expectation - this error message just says 'expected false to equal true' which is hard to understand in a hurry.

**Comment by pivotal-jamie-klassen** [comment]
@Lindsayauchin @matthewpereira here's the error page when you're trying to `fly login` with Safari to a TLS-secured Concourse server. Maybe you folks can give some feedback about styles and copy?
Screen Shot 2019-09-16 at 10 52 00 AM

**Comment by pivotal-jamie-klassen** [comment]
lindsays notes:

"however, your token could not be sent to fly because your browser blocked the attempt.

send token to fly directly < [make into button]

[more space]

if that fails, you will need to copy the token to your clipboard, return to fly, and paste your token into the prompt.

[copy token to clipboard]"

**Comment by Lindsayauchin** [comment]
Screen for reference to the above note ^^

Screen Shot 2019-09-18 at 2 23 39 PM

**Comment by pivotal-jamie-klassen** [comment]
@Lindsayauchin
Screen Shot 2019-09-18 at 4 29 52 PM

**Pull request merged**